CMakeLists.txt

```diff
@@ -80,43 +91,42 @@ endif()
 
 configure_file(config.h.in config.h)
 
+# Configure threads
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+
 # Support functionality.
 
-function(add_compile_flag value)
-  message(STATUS "Building with ${value}")
-  foreach(variable CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
-    set(${variable} "${${variable}} ${value}" PARENT_SCOPE)
-  endforeach(variable)
-endfunction()
+# Use a property to emulate a global variable.
+set_property(GLOBAL PROPERTY binaryen_1p_targets binaryen)
```

Comment:
can't a setting work as well? (I've never seen cmake properties used before, and at a glance at the usage and docs, it's not obvious to me what is added here, since this is global)

Reply:
I'm not sure what you mean by a "setting." AFAIK cmake has no such concept. I can't use a normal variable because they can only be set in current or immediate parent scopes, so there's no way for `binaryen_add_executable` calls from subdirectories to modify a shared list of targets. The only other kind of "variable" cmake has is properties. This particular property is not stored in the cache because there's no reason for the user to have visibility into it.

Reply:
Moving the flags management one scope down wouldn't be sufficient to use a normal variable here because executables are added several scopes down (and also in test/gtest/, which is not under src/). Maybe you have a larger refactoring in mind? I would strongly prefer to declare Mission Accomplished on this and not try to get a very different architecture working, given how difficult it is to make correct changes here.

Reply:
Also, FWIW `EMSCRIPTEN` and `MSVC` are normal CMake variables. They are visible in child scopes, but mutations performed in those scopes would be done to copies of the variables and would not be visible back in the parent or in sibling scopes. (Yes this is awful.)

Reply:
> Moving the flags management one scope down wouldn't be sufficient to use a normal variable here

Why do we need a normal variable here? The goal of the PR IIUC is to apply custom flags only to 1P code. If we had a directory structure like this:
```
src/        ;; all of this is 1P code
thirdparty/ ;; all of this is 3P code
```
then wouldn't the problem be trivial? We would apply custom flags inside `src/` and that's it. Sorry if I'm missing something obvious, but this seems simple to do (so likely I am confused).